### PR TITLE
feat(sub-category):associate subcategory with category

### PR DESCRIPTION
**Pull request by iankigen**
**What does this PR do?**
Adds an association between category and the sub-category models

**Description of work done?**
- Write tests for the association
- Add a category foreign key in the sub-category model

**How can this be manually tested?**
> $ python manage.py test

**Relevant pivotal tracker stories?**
[#155322119](https://www.pivotaltracker.com/n/projects/2146417/stories/155322119)

### CODE REVIEW

**Comment by iankigen** [comment]
@egichuri @jmnyarega @HawiCaesar  Kindly review

File: .gitignore [hunk 1]
```diff
@@ -103,3 +103,4 @@ ENV/
 
 # mypy
 .mypy_cache/
+.idea
```
Comment (jmnyarega) [inline .gitignore]: Oops

Comment (HawiCaesar) [inline .gitignore]: This is Pycharm. I don't think the whole team uses it. You don't have to push this

**Comment by HawiCaesar** [comment]
LGTM

**Pull request merged**